src/jit/lsra.cpp

```diff
@@ -5724,13 +5724,14 @@ regNumber LinearScan::allocateBusyReg(Interval* current, RefPosition* refPositio
             }
         }
 
-        LsraLocation nextLocation = assignedInterval->getNextRefLocation();
+        RefPosition* nextRefPosition = assignedInterval->getNextRefPosition();
+        LsraLocation nextLocation    = assignedInterval->getNextRefLocation();
 
         // We should never spill a register that's occupied by an Interval with its next use at the current location.
         // Normally this won't occur (unless we actually had more uses in a single node than there are registers),
         // because we'll always find something with a later nextLocation, but it can happen in stress when
         // we have LSRA_SELECT_NEAREST.
-        if ((nextLocation == refLocation) && !refPosition->isFixedRegRef)
+        if ((nextLocation == refLocation) && !refPosition->isFixedRegRef && nextRefPosition->RequiresRegister())
```

Comment:
Over all the tests, frameworks & benchmarks there is a very small (48 byte) code size increase.
For the benchmarks the net is a very small overall (218) byte code size reduction:

Improvements:
Platform.Graphics.Typography.TrueType.ScanConverter:CalcSpline() goes from  3205 to  2953, diff of 252 (7.86%)
IDEAEncryption:cipher_idea() goes from  1135 to  1124, diff of 11 ( 0.97%)

Regressions:
Platform.Graphics.Typography.TrueType.LocalGraphicState:IUPx() goes from  2156 to  2186, diff of 30 (1.37%)
Platform.Graphics.Typography.TrueType.ScanConverter:MeasureGlyph() goes from  2566 to  2581, diff of 15 (0.58%)